issue_14/script.js

```diff
@@ -0,0 +1,92 @@
+function startLoading() {
+    const loadingImg = document.createElement('img');
+    loadingImg.src = 'loading-circle.gif';
+    loadingImg.alt = 'ローディング画像';
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function removeLoading() {
+    document.getElementById('js-loading').remove();
+}
+
+function renderList(resolvedData) {
+    const ul = document.getElementById('js-ul');
+    const fragment = document.createDocumentFragment();
+    for (const item of resolvedData) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+
+        a.href = item.a;
+        a.textContent = item.text;
+        img.src = item.img;
+        img.alt = item.alt;
+
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+    ul.appendChild(fragment);
+}
+
+function renderErrorMessage(messageText) {
+    document.getElementById('js-ul').textContent = messageText;
+}
+
+async function fetchData() {
+    try {
+        const url = 'https://api.json-generator.com/templates/NluaaELSLhVe/data?access_token=ambjpfxjl00e50wa639kwndq1ofq3iuykdrv98ge';
+        const response = await fetch(url);
+        if (!response.ok) {
+            console.error(`${response.status}:${response.statusText}`);
+            renderErrorMessage(`${response.status}:${response.statusText}`);
+        }
+        const result = await response.json();
+        return result.data;
+    } catch (error) {
+        console.error(error);
+        renderErrorMessage(`${error.name}:${error.message}`);
+    }
+}
+
+async function init(num) {
+    startLoading();
+    let listData;
+    try {
+        console.log(num);
+        listData = await fetchData();
+        if (!listData) {
+            return;
+        }
+    } catch (error) {
+        console.error(error.message);
+        renderErrorMessage(error.message);
+    } finally {
+        removeLoading();
+    }
+    if (!listData.length) {
+        console.log('データがありません');
+        renderErrorMessage('データがありません');
+    }
+    renderList(listData);
+}
+
+const modalBtnWrapper = document.getElementById('js-modal-btn__wrapper');
+const modalWrapper = document.getElementById('js-modal__wrapper');
+const requestBtn = document.getElementById('js-request-btn');
+const modalButtonElements = document.getElementsByClassName('js-modal-btn');
+
+for (const modalButtonElement of modalButtonElements) {
+    modalButtonElement.addEventListener('click', () => {
+        modalBtnWrapper.classList.toggle('hide');
+        modalWrapper.classList.toggle('show');
+    });
+}
```

Comment:
This modal switching was new to me! Interesting!

IMO
I thought if I made this into a function and gave it a name, it would be easier to see what was being done.

Reply:
Thank you for your comment.
It would certainly be easier to read this section if it were summarized in a function.
I have corrected the part you pointed out. 
https://github.com/yoshikawa-akr/js-lessons/pull/14/commits/16d69b01dd535259f4977e998d85a5d00523df34